packages/metal-incremental-dom/src/IncrementalDomRenderer.js

```diff
@@ -218,10 +222,25 @@ class IncrementalDomRenderer extends ComponentRenderer.constructor {
 	 * @param {!Component} component
 	 */
 	update(component) {
-		if (this.shouldUpdate(component, getChanges(component))) {
+		const changes = getChanges(component);
+		if (this.shouldUpdate(component, changes)) {
+			this.willUpdate_(component, changes);
+
 			this.patch(component);
 		}
 	}
+
+	/**
+	 * Invokes component's "willUpdate" lifecycle method if applicable.
+	 * @param {!Component} component
+	 * @param {Object} changes
+	 */
+	willUpdate_(component, changes) {
+		if (!component.willUpdate || !component.wasRendered || !changes) {
```

Comment:
See comment above... if we want to follow the `render` approach, we could proved a wrapping API for it like `component.informWillUpdate`. I'm fine keeping it here too :)

Reply:
Also, from the calling code, it might not be needed to check for `!changes`, since it is always invoked within something that used `shouldUpdate(..., changes)` which already does that check?

Reply:
Hey @jbalsas 

In that case we might want to do the same for `shouldUpdate`, right now it just checks for a `component.shouldUpdate` method from the [renderer](https://github.com/metal/metal.js/blob/master/packages/metal-incremental-dom/src/IncrementalDomRenderer.js#L191-L193). What do you think?

Reply:
Yup, we probably should! 👍